x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/utils.ts

```diff
@@ -186,6 +186,8 @@ export async function getVisTypeFactory(lens: LensPublicStart) {
 export async function isCompatibleVisualizationType(chartInfo: ChartInfo) {
   return (
     chartInfo.visualizationType === COMPATIBLE_VISUALIZATION &&
+    // @ts-expect-error esql is missing in the type
+    chartInfo.query.esql === undefined &&
```

Comment:
This change looks good, but I can't remember - did we ever try and check that the visualization has a time based axis? For example, you also see the same warning for this viz on the dashboard which comes with the web logs sample data set:

<img width="560" alt="image" src="https://github.com/elastic/kibana/assets/7405507/e466b308-51f7-4085-8e61-ce277bdede93">

Reply:
We perform this check when opening the flyout, which is what you're seeing there, not during the `isCompatible` check which determines whether to show the action menu item.

When this feature was originally added we decided that only a simple check should be performed in the `isCompatible` function to see if each layer is of type data and that chart is a compatible type.

Originally also we would hide the action menu item if no layers were compatible, but I believe it was decided it would be better to show the menu item to show that anomaly detection is a possible feature and when the flyout opens we can list why each layer isn't compatible.
but here was argument that it's annoying to click something only to find out you can't use it.

We could change this to do a deeper inspection for each layer during the `isCompatible` check and not to show the action menu item if no layers are compatible. But I think that should be done in a different PR.

Reply:
Thanks for the clarification. Makes sense. I think it probably is worth enhancing the `isCompatible` check in a separate PR to hide the action if the panel contains no compatible layers.